### PR TITLE
Add support for gcp sql proxy for hydra

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Add new ory subchart - gcp proxy
- Update production profile
- Update docs
- Minor fixes and refactor

**Related issue(s)**

- Implementes #7053

### CODE REVIEW

File: installation/resources/installer-cr.yaml.tpl [hunk 1]
```diff
@@ -24,16 +24,16 @@ spec:
       namespace: "kyma-installer"
     - name: "istio-kyma-patch"
       namespace: "istio-system"
-    - name: "knative-serving-init"
-      namespace: "knative-serving"
-    - name: "knative-serving"
-      namespace: "knative-serving"
+    # - name: "knative-serving-init"
```
Comment (strekm) [inline installation/resources/installer-cr.yaml.tpl]: why changes in this files?

File: resources/ory/charts/gcloud-sqlproxy/OWNERS [hunk 2]
```diff
@@ -0,0 +1,5 @@
+approvers:
```
Comment (strekm) [inline resources/ory/charts/gcloud-sqlproxy/OWNERS]: i think this file should be removed

File: installation/resources/installer-cr.yaml.tpl (hunk 1 above)
Comment (Demonsthere) [reply installation/resources/installer-cr.yaml.tpl]: That was a change for local testing, should not have been commited :D

File: docs/security/05-05-production-profile.md [hunk 3]
```diff
@@ -5,35 +5,104 @@ type: Configuration
...
 
-Alternatively, you can use a compatible external database to store the registered client data. To use an external database, you must create a Kubernetes Secret with the database password in the same Namespace as the Hydra OAuth2 server. The details of the external database are passed through these parameters of the production profile override:
+In order to use the database, the following configuration parameters have to be supplied:
+
+**General settings:**
+
+| Parameter | Description | Required value |
+|----------|-----------| :---: |
+| **global.ory.hydra.persistence.enabled** | Defines if Hydra should use the `in-memory` or `database` mode of operation | `true` |
+| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use an  database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `true` |
```
Comment (strekm) [inline docs/security/05-05-production-profile.md]:
```suggestion
| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a custom database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `true` |
```

File: docs/security/05-05-production-profile.md [hunk 4]
```diff
@@ -5,35 +5,104 @@ type: Configuration
...
+#### Custom database
+
+Alternatively, you can use a compatible, custom database to store the registered client data. To use a database, you must create a Kubernetes Secret with the database password as an override for the Hydra OAuth2 server. The details of the database are passed through these parameters of the production profile override:
+
+**General settings:**
+
+| Parameter | Description | Required value |
+|----------|------| :---: |
+| **global.ory.hydra.persistence.enabled** | Defines if Hydra should use the `in-memory` or `database` mode of operation | `true` |
+| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `false` |
```
Comment (strekm) [inline docs/security/05-05-production-profile.md]:
```suggestion
| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a custom database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `false` |
```

File: docs/security/05-05-production-profile.md [hunk 5]
```diff
@@ -5,35 +5,104 @@ type: Configuration
...
+
+#### Google Cloud SQL
+The Cloud SQL is a provider supplied and maintained database, which requires a special proxy deployment in order to provide a secured connection. In Kyma we provide a [pre-installed](https://github.com/rimusz/charts/tree/master/stable/gcloud-sqlproxy) deployment, which requires the following parameters in order to operate:
+
+**General settings:**
+
+| Parameter | Description | Required value |
+|----------|------| :---: |
+| **global.ory.hydra.persistence.enabled** | Defines if Hydra should use the `in-memory` or `database` mode of operation | `true` |
+| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `false` |
```
Comment (strekm) [inline docs/security/05-05-production-profile.md]:
```suggestion
| **global.ory.hydra.persistence.postgresql.enabled** | Defines if Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a custom database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `false` |
```

File: docs/security/05-05-production-profile.md [hunk 6]
```diff
@@ -5,35 +5,104 @@ type: Configuration
...
-| **global.ory.hydra.persitance.enabled** | Defines if Hydra should use the `in-memory` or `database` mode of operation |
-| **data.global.ory.hydra.persitance.user** | Specifies the name of the user with permissions to access the database. |
-| **data.global.ory.hydra.persitance.secretName** | Specifies the name of the Secret in the same Namespace as Hydra that stores the database password. |
-| **data.global.ory.hydra.persitance.secretKey** | Specifies the name of the key in the Secret that contains the database password. |
-| **data.global.ory.hydra.persitance.dbUrl** | Specifies the database URL. For more information, read [this](https://github.com/ory/hydra/blob/master/docs/config.yaml) document. |
-| **data.global.ory.hydra.persitance.dbName** | Specifies the name of the database saved in Hydra. |
-| **data.global.ory.hydra.persitance.dbType** | Specifies the type of the external database. The supported protocols are `postgres`, `mysql`, `cockroach`. Follow [this](https://github.com/ory/hydra/blob/master/docs/config.yaml) link for more information. |
+>>**NOTE:** When using any kind of custom database (gcloud, or self-maintained), it is important to set the
```
Comment (majakurcius) [inline docs/security/05-05-production-profile.md]: Aren't we missing something here?

File: docs/security/05-05-production-profile.md [hunk 7]
```diff
@@ -90,9 +161,10 @@ You can deploy a Kyma cluster with the Hydra OAuth2 server configured to use the
...
         kyma-project.io/installation: ""
     data:
-      postgresql.enabled: "true"
+      global.ory.hydra.persistence.enabled: "true"
+      global.ory.hydra.persistence.postgresql.enabled: "true"
+      global.ory.hydra.persistence.gcloud.enabled: "false"
       hydra.hydra.autoMigrate: "true"
-      global.ory.hydra.persitance.enabled: "true"
     EOF
     ```
```
Comment (piotrmsc) [inline docs/security/05-05-production-profile.md]: Please add example for gcp sql which will be displayed as a 3rd tab/option as an example

File: docs/security/05-05-production-profile.md [hunk 8]
```diff
@@ -5,35 +5,105 @@ type: Configuration
...
+| Parameter | Description | Required value |
+|----------|------| :---: |
+| **global.ory.hydra.persistence.enabled** | Defines whether Hydra should use the `database` mode of operation. If false, it uses the `in-memory` mode. | `true` |
+| **global.ory.hydra.persistence.postgresql.enabled** | Defines whether Hydra should initiate the deployment of an in-cluster database. Set to `false` to use a self-provided database. If set to `true`, Hydra always uses an in-cluster database and ignores the custom database details. | `false` |
+| **hydra.hydra.autoMigrate** | Enables Hydra auto-migration feature, which prepares the database. | `true` | 
+| **hydra.hydra.config.secrets.system** | Sets the system encryption string for Hydra. | An at least 16 characters long alphanumerical string | 
+| **hydra.hydra.config.secrets.cookie** | Sets the cookie session encryption string for Hydra. | An at least 16 characters long alphanumerical string |
+
+**Database settings:**
+
```
Comment (piotrmsc) [inline docs/security/05-05-production-profile.md]: Currently, the document is weirdly formatted, take a look : https://deploy-preview-7130--kyma-project-docs-preview.netlify.com/components/security/#configuration-o-auth2-server-production-profile-google-cloud-sql

File: docs/security/05-05-production-profile.md (hunk 6 above)
Comment (Demonsthere) [reply docs/security/05-05-production-profile.md]: Commited without saving, my bad :)

**Pull request merged**